lua/jsonpath.lua

```diff
@@ -88,4 +88,80 @@ M.get = function()
   return path
 end
 
+local function get_yaml_path()
+  if not parsers.has_parser() then
+    return ""
+  end
+
+  local current_node = ts_utils.get_node_at_cursor()
+  if not current_node then
+    return ""
+  end
+
+  local accessors = {}
+  local node = current_node
+
+  while node do
+    local accessor = ""
+
+    if node:type() == "block_mapping_pair" then
+      local key_node = unpack(node:field("key"))
+      local key = get_node_text(key_node)
+
+      if key and starts_with_number(key) or contains_special_characters(key) then
```

Comment:
Currently, this doesn't handle JSON inside YAML.

For instance, with my cursor on

```yaml
json_map: { |"key": "value" }
```

It returns `json_map`. I'd expect it to return `json_map.key`.

I could make it work by changing also handle the node type `flow_pair`

```suggestion
    if node:type() == "block_mapping_pair" or node:type() == "flow_pair" then
      local key_node = unpack(node:field("key"))
      local key = get_node_text(key_node)

      if key and starts_with_number(key) or contains_special_characters(key) then
```

But then the result would be also unexpected, `json_map[""key""]` instead of `json_map.key`. So we should handle the node `double_scalar_key` separately.

The node `string_scalar` seems to work fine, for instance in this line:

```yaml
and quotes are optional: { |key: [ 3, 2, 1, takeoff ] }
```